R/transformAbundance.R

```diff
@@ -0,0 +1,400 @@
+                                                          scale = scale,
+                                                          target = target)
+
+              # Assign transformed table to assays
+              assay(x, name) <- transformed_table
+
+              return(x)
+
+          }
+)
+
+#' @rdname transformAbundance
+#' @export
+setGeneric("getTransformAbundance", signature = c("x"),
+           function(x,
+                    abund_values = "counts",
+                    transform = "identity",
+                    name = transform,
+                    pseudocount = FALSE,
+                    scale = 1,
+                    target = "samples")
+               standardGeneric("getTransformAbundance"))
+
+
+#' @rdname transformAbundance
+#' @export
+setMethod("getTransformAbundance", signature = c(x = "SummarizedExperiment"),
+          function(x,
+                   abund_values = "counts",
+                   transform = "identity",
+                   name = transform,
+                   pseudocount = FALSE,
+                   scale = 1,
+                   target = "samples"){
+
+              # Get object with transformed table
+              x <- transformAbundance(x, abund_values, transform, name, pseudocount, scale, target)
+
+              # Get the table
+              mat <- assay(x, name)
+
+              return(mat)
+          }
+)
+
+###########################HELP FUNCTIONS####################################
+
+
+# Chooses which transformation function is applied
+.get_transformed_table <- function(assay, transform, pseudocount, scale, target){
+
+    # If "pseudocount" is TRUE or over 0 or transform is log10p, add pseudocount
+    if(!pseudocount==FALSE || transform=="log10p"){
+        if(is.logical(pseudocount)){
+            # Add 1 as a pseudo count
+            assay <- assay + 1
+            warning("Transform was calculated with pseudocount value 1")
+        } else{
+            # When user have specified pseudocount, add pseudocount as a pseudocount
+            assay <- assay + pseudocount
+        }
+    }
+
+    # Multiply values with scale. By default, scale is 1, so no changes are made
+    assay <- assay * scale
+
+    # Function is selected based on the "transform" variable
+    FUN <- switch(transform,
+                  relabundance = .get_relabundance_table,
+                  Z = .get_z_table,
+                  log10 = .get_log10_table,
+                  log10p = .get_log10p_table,
+                  hellinger = .get_hellinger_table,
+                  identity = .get_identity_table,
+                  clr = .get_clr_table)
+
+    # Does the function call, arguments are "assay" abundance table and "pseudocount"
+    do.call(FUN,
+            list(assay = assay,
+                 target = target))
+}
+
+.get_relabundance_table <- function(assay, target){
+
+    # Calculates the relative abundances
+    mat <- sweep(assay, 2, colSums(assay), "/")
+    return(mat)
+}
```

Comment:
this exists already in `relAbundanceCounts`.

Reply:
Should transform=relabundance still be a choice here in transformAbundance? I changed this part so that row 272 (shown above) is changed to use internal function .calc_rel_abund

Reply:
I think it should be an option.

Reply:
Ok